source/agora/flash/OnionPacket.d

```diff
@@ -512,3 +512,49 @@ public struct OnionError
     /// Error code
     ErrorCode err;
 }
+
+/*******************************************************************************
+
+    Obfuscate/deobfuscate an OnionError with the given secret.
+
+    Params:
+        error = Error to obfuscate/deobfuscate
+        secret = Shared secret
+
+    Returns:
+        Obfuscated/deobfuscated OnionError
+
+*******************************************************************************/
+
+public OnionError obfuscate (OnionError error, Point secret) @trusted
+{
+    import libsodium.crypto_stream_chacha20;
+
+    const key = hashFull(secret);
+    assert(key[].length >= crypto_stream_chacha20_KEYBYTES);
```

Comment:
Is this technique with chacha20 described in the LN RFC? From my brief glance at the RFC I thought the same technique would be used as the one for onion packets.

It's fine to use chacha though.

Reply:
RFC does not specify any algorithm but the implementation uses chacha. 

https://github.com/lightningnetwork/lightning-onion/blob/3c8c8d0b89ea82958d1fae0657e3eaf5f1267b5f/crypto.go#L138-L150